helm-app-operator/pkg/helm/controller/controller.go

```diff
@@ -0,0 +1,68 @@
+package controller
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/operator-framework/helm-app-operator-kit/helm-app-operator/pkg/helm/installer"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	crthandler "sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// WatchOptions contains the necessary values to create a new controller that
+// manages helm releases in a particular namespace based on a GVK watch.
+type WatchOptions struct {
+	Installer installer.Installer
+	Namespace string
+	GVK       schema.GroupVersionKind
+	// StopChannel is used to deal with the bug:
+	// https://github.com/kubernetes-sigs/controller-runtime/issues/103
+	StopChannel <-chan struct{}
+}
+
+// Add creates a new helm operator controller and adds it to the manager
+func Add(mgr manager.Manager, options WatchOptions) {
+	hor := &helmOperatorReconciler{
+		Client:    mgr.GetClient(),
+		GVK:       options.GVK,
+		Installer: options.Installer,
+	}
+
+	// Register the GVK with the schema
+	mgr.GetScheme().AddKnownTypeWithName(options.GVK, &unstructured.Unstructured{})
+	metav1.AddToGroupVersion(mgr.GetScheme(), options.GVK.GroupVersion())
+
+	// Create new controller-runtime controller and set the controller to watch this GVK.
+	c, err := controller.New(fmt.Sprintf("%v-controller", strings.ToLower(options.GVK.Kind)), mgr, controller.Options{
+		Reconciler: hor,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	u := &unstructured.Unstructured{}
+	u.SetGroupVersionKind(options.GVK)
+	if err := c.Watch(&source.Kind{Type: u}, &crthandler.EnqueueRequestForObject{}); err != nil {
+		log.Fatal(err)
+	}
+
+	// Emit events once per minute regardless of detected changes.
+	r := newReconcileLoop(time.Minute*1, options.GVK, mgr.GetClient())
```

Comment:
I'm wondering if we really need this (both here and in the ansible-operator).
@shawn-hurley Can you comment on why we can't just set the resync option for the manager(which sets it for the cache)? 
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/manager.go#L91-L95
```
mgr, err := manager.New(cfg, manager.Options{Namespace: namespace, SyncPeriod: time.Minute*1})
```

Reply:
Discussed offline.
For our use case the manager's resync is ***almost*** equivalent to the reconcile loop.
https://github.com/kubernetes-sigs/controller-runtime/pull/88#issuecomment-408546638
So we can use either.

However the recently added [RequeueAfter](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/reconcile/reconcile.go#L30-L31) is probably the best way to periodically re-queue and re-trigger the reconcile for each CR.

Reply:
Agreed. I'll make the change and verify expected behavior.